python/ray/experimental/channel/common.py

```diff
@@ -196,17 +197,23 @@ def write(self, value: Any) -> None:
 
         Args:
             value: The value to write.
+            timeout: The maximum time in seconds to wait to write the value.
+                If specified, the value must be a positive number.
```

Comment:
Better if we can allow:
- -1 for infinite timeout
- 0 means try once but throw error if we would block

Reply:
We already have `None` for infinite timeout. Added support for 0, and will constrain the value to be non-negative.